### DISCUSSION TITLE
[NEED HELP] MongoEngine and FastAPI errors

### DISCUSSION THREAD

**Discussion by naheedroomy** [fastapi/fastapi#3998] [Questions] [answered]
Hey guys, was developing on Flask for a while before I came across FastAPI and absolutely love it.

I have a small question though. I very much prefer MongoEngine over other modules like PyMongo and Motor because of its ORM like nature and I would prefer to have an answer related to MongoEngine.

So this is my question. I've connected my simple user management application to a db,

```
from mongoengine import connect

connect(host="mongodb://127.0.0.1:27017/pythonmongodb")
```

I've also defined the model of the user, and the document model as well.

```

class User(BaseModel):
    id: Optional[str]
    username: str
    email: str
    password: str


class Users(Document):
    username: StringField(required=True)
    password = StringField(required=True, min_length=4)
    email = EmailField(required=True, unique=True)
```
Now, I'm trying to add a user to this, however am getting the error stating "

`mongoengine.errors.FieldDoesNotExist: The fields "{'username'}" do not exist on the document "Users"`

The code that I'm using for the post request is:

```
@user.post('/users', response_model=User, tags=["users"])
async def create_user(user: User):
    new_user = Users(**user.dict()).save()
    id = new_user.id

    return id
```

As for my database, I do have a database called "pythonmongodb" which has a collection called "Users"

I'm still new to this whole API thing, so please do advice me in what you think I'm doing wrong!

**Comment by negadive** [comment, marked as answer]
i think u didnt actually define the username attribute

```python
class Users(Document):
    username: StringField(required=True)
```

which should be
```python
class Users(Document):
    username = StringField(required=True)
```
